### PR TITLE
dark css theme version 0.1

### PR DESCRIPTION
**Pull request by savez**
Created dark theme version 0.1 

reference issue https://github.com/goksan/statusnook/issues/17

### CODE REVIEW

**Comment by goksan** [comment]
Hi @savez, thanks for the PR

How’s it going? Do you have any details or images to share?

Let me know if you need any help

**Comment by savez** [comment]
Screenshot 2024-09-03 at 07 03 22
Screenshot 2024-09-03 at 07 03 18
Screenshot 2024-09-03 at 07 03 12
Screenshot 2024-09-03 at 07 03 05
Screenshot 2024-09-03 at 07 02 59

**Comment by savez** [comment]
I was thinking that perhaps one could use the css function light-dark() (https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark) so as to have only one CSS and use the colour-scheme proprity to make the switch.

what do you think?

**Comment by goksan** [comment]
Sure, that sounds like it works

**Comment by savez** [comment]
Good, I implement it and update PR

**Comment by savez** [comment]
@goksan I change it.

**Comment by goksan** [comment]
Thank you! If you're done with it I'll try and take a proper look later this week, and experiment with some of the colours

Thanks again

**Comment by goksan** [comment]
Did you intentionally commit sizing & font changes?

**Comment by savez** [comment]
Yes, I think that this UI are good for more screen size.

**Comment by savez** [comment]
I think a button for switching the theme should be added in the interface.
what do you think?

**Comment by goksan** [comment]
Would be helpful for users that want to override their OS default

**Comment by savez** [comment]
@goksan added toggle button

**Comment by goksan** [comment]
Ok, have you made all your changes?

**Comment by savez** [comment]
I think yes

**Comment by goksan** [comment]
Are these changes generated by AI?

**Comment by savez** [comment]
> Are these changes generated by AI?

no, it is my development

File: static/main.css [hunk 1]
```diff
@@ -1,23 +1,98 @@
+:root{
+  /* light styles here */
+  color-scheme: var(--color-scheme, light);
```
Comment (goksan) [inline static/main.css]: `color-scheme` is duplicated, this is 1

File: static/main.css [hunk 2]
```diff
@@ -1,23 +1,98 @@
...
+    
+  &:has(#theme-select option[value="dark"]:checked) {
+    --color-scheme: dark;
+  }
+
+  &:has(#theme-select option[value="light"]:checked) {
+    --color-scheme: light;
+  }
+
+    color-scheme:  var(--color-scheme, light);
```
Comment (goksan) [inline static/main.css]: `color-scheme` is duplicated, this is 2

File: static/main.css [hunk 3]
```diff
@@ -1,23 +1,98 @@
+:root{
+  /* light styles here */
```
Comment (goksan) [inline static/main.css:2]: is this supposed to be here?

File: static/main.css [hunk 4]
```diff
@@ -501,7 +578,7 @@ label {
 .nav > a {
     font-size: 1.6rem;
     font-weight: 500;
-    color: #969696;
+        color: light-dark(#606060,#969696);
```
Comment (goksan) [inline static/main.css]: there's a bunch of places where extra indentation has been added

**Review by goksan — COMMENTED** [review]
Why does the light theme look like this?
Screenshot 2024-09-05 at 11-17-11 Services

This does not default to light/dark based on the OS theme.

The selected theme is lost when you navigate to any other page. Maybe you didn't get around to this, but you haven't provided any detail to suggest this.

I'm not a fan of the sizing changes, and those weren't initially mentioned until I spotted them. Same goes for the font changes.

I'm not going to merge this in its current state. I appreciate you have taken the time to make these current changes, but I also have to take time to review them and a lot of key info has been missing.

**Comment by savez** [comment]
I check your comment and I will implement the necessary changes. thank you!

**Comment by savez** [comment]
> Why does the light theme look like this? Screenshot 2024-09-05 at 11-17-11 Services
> 
> This does not default to light/dark based on the OS theme.
> 
> The selected theme is lost when you navigate to any other page. Maybe you didn't get around to this, but you haven't provided any detail to suggest this.
> 
> I'm not a fan of the sizing changes, and those weren't initially mentioned until I spotted them. Same goes for the font changes.
> 
> I'm not going to merge this in its current state. I appreciate you have taken the time to make these current changes, but I also have to take time to review them and a lot of key info has been missing.

I'm sorry for the bugs. I will implement the necessary changes

For the font I not change it

For the font-site I changed it because I had thought it was better for the view of the data

**Comment by savez** [comment]
Hi,
I am sorry for the error into prev PR.

I don't seen that you use HTMX for system routing and my implementation didn’t work.

I correct it and I added `afterSwap` event for manage theme selector.

I hope that this implementation it is ok for you.


Bye

**Comment by goksan** [comment]
I've had a quick look and issues remain from my prior comments.

> Why does the light theme look like this?

It still looks like this?

> This does not default to light/dark based on the OS theme.

This still seems to be the case.

>The selected theme is lost when you navigate to any other page. Maybe you didn't get around to this, but you haven't provided any detail to suggest this.

Navigating backwards or forwards between pages while in dark mode causes the page you land on to be displayed in light mode.

> I'm not a fan of the sizing changes, and those weren't initially mentioned until I spotted them. Same goes for the font changes.

> For the font I not change it

To be clear I was referring to font size & weight.

**Comment by savez** [comment]
Hello,
the font (weights etc.) I changed them because it seems more readable to me.

I'll post a video so you can see that browsing remains the dark theme.

The theme type is saved in localStorage and stays saved there.

only the login is in light theme but by choice. It is not nice to give in login the ability to change theme.

https://github.com/user-attachments/assets/3dc8b174-94e1-4f03-866c-b3c8d3d75e91

**Comment by goksan** [comment]
> Hello, the font (weights etc.) I changed them because it seems more readable to me.

To be clear - I'm not looking to change that at this time.

I think you've changed the background and icon colours, surely you can agree this isn't an improvement?




> I'll post a video so you can see that browsing remains the dark theme.

To be clear - you need to navigate backwards and forwards using the back or forward button in your __browser__ to reproduce the issue. I've included a video demonstrating this.

https://github.com/user-attachments/assets/0a51ae49-0ef2-4dfc-b170-268299f58c9c


> It is not nice to give in login the ability to change theme.

You probably don't need to change theme on the login screen, I agree, but I would expect it to use the theme I've selected.

**Comment by savez** [comment]
ok, thanks epr for the clarification. I will check for the theme and browser navigation keys

I'm sorry

**Comment by goksan** [comment]
Thank you for your time, but I won't be merging this PR.

**Comment by savez** [comment]
@goksan 

HTMX is probably the problem. 

I have made changes to handle:
- system preference
- navigation with browser keys

I then found this plugin “https://unpkg.com/dark-mode-toggle” which works and seems much more convenient for management.

I'll go over a PR if you want.

**Comment by goksan** [comment]
I don't think such a plugin is necessary here

**Comment by savez** [comment]
The plugin was an example for the best way for implamentation

Bye

**Pull request closed without merging**